app/scripts/controllers/mmi-controller.ts

```diff
@@ -790,23 +790,25 @@ export default class MMIController extends EventEmitter {
     const isCustodial = Boolean(accountDetails);
     const updatedMsgParams = { ...msgParams, deferSetAsSigned: isCustodial };
 
-    if (req.method.includes('eth_signTypedData')) {
+    if (
+      req.method === 'eth_signTypedData' ||
+      req.method === 'eth_signTypedData_v3' ||
+      req.method === 'eth_signTypedData_v4'
+    ) {
       return await this.signatureController.newUnsignedTypedMessage(
         updatedMsgParams as PersonalMessageParams,
         req as OriginalRequest,
         version,
         { parseJsonData: false },
       );
-    } else if (req.method.includes('personal_sign')) {
+    } else if (req.method === 'personal_sign') {
       return await this.signatureController.newUnsignedPersonalMessage(
         updatedMsgParams as PersonalMessageParams,
         req as OriginalRequest,
       );
     }
-    return await this.signatureController.newUnsignedMessage(
```

Comment:
@adonesky1, the only thing I'm afraid of in this PR is the removal of this `newUnsignedMessage` for the custodian accounts. Let me talk with the team first. cc @zone-live @shane-t

Reply:
I'm not concerned about it, because it willy only be true if the method is `eth_sign` based on the other conditions in the block.